src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+    private static Scanner scanner = new Scanner(System.in);
+    private static String horizontalLine = "************************";
+    private static ArrayList<Task> taskArr;
+
+    private static Storage storage = new Storage();
+
     public static void main(String[] args) {
+        // Load list from storage
+        taskArr = storage.loadList();
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        System.out.println(horizontalLine);
+        System.out.println("HELLO! I'M DUKE!");
+        System.out.println("HOW CAN I HELP?");
+        System.out.println("NOTE! USE THIS FORMAT FOR DATES: dd/MM/yyyy HH:mm");
+        System.out.println(horizontalLine);
+        readInput();
     }
+
+    private static void updateListStorage() {
+        storage.storeList(taskArr);
+    }
+
+    private static void readInput() {
+        String input = scanner.nextLine();
+        try {
+            if (input.equals("bye")) {
+                showBye();
+            } else if (input.equals("list")) {
+                showList();
+            } else if (input.startsWith("mark")) {
+                showMark(input);
+            } else if (input.startsWith("unmark")) {
+                showUnmark(input);
+            } else if (input.startsWith("todo")) {
+                addTodo(input);
+            } else if (input.startsWith("deadline")) {
+                addDeadline(input);
+            } else if (input.startsWith("event")) {
+                addEvent(input);
+            } else if (input.startsWith("delete")) {
+                deleteTask(input);
+            } else {
+                throw new DukeException("I DON'T UNDERSTAND THAT!");
+            }
+        } catch (DukeException e) {
+            showUnknown(e.getMessage());
+        }
+    }
+
+    private static void showUnknown(String errorMsg) {
+        System.out.println(horizontalLine);
+        System.out.println(":( SORRY! " + errorMsg);
+        System.out.println(horizontalLine);
+        readInput();
+    }
+
+    private static void addTodo(String input) throws DukeException {
+        // Error handling
+        if (input.length() <= 5) {
+            throw new DukeException("TODO NEEDS A DESCRIPTION!");
+        }
+
+        // Get to-do name
+        String todo = input.substring(5);
```

Comment:
It might more sense to name `todo` as `description` or `taskDesc`, since that's the field name in `Task` (maybe also for the `taskName`s in `addDeadline` and `addEvent`)

---

Also, a pretty cool trick with `split` is that you can pass a `limit` as the second parameter, so you can do this to extract just the first word:

```java
String input = "todo trash the trash";
String[] arr = input.split(" ", 2); // only split to 2 elements
arr[0]; // "todo"
arr[1]; // "trash the trash"
```

So you can do something like this if you want to:

```suggestion
        String description = input.split(" ", 2)[1];
```

It's a bit longer, but at least you won't have to hardcode the number of characters.

(I would actually recommend doing the splitting after line 35 so that you can separate the command and arguments immediately)